gcp-observability/src/main/java/io/grpc/gcp/observability/ObservabilityConfigImpl.java

```diff
@@ -73,96 +88,151 @@ void parse(String config) throws IOException {
-      rawList = JsonUtil.getList(config, "event_types");
-      if (rawList != null) {
-        List<String> jsonEventTypes = JsonUtil.checkStringList(rawList);
-        ImmutableList.Builder<EventType> eventTypesBuilder = new ImmutableList.Builder<>();
-        for (String jsonEventType : jsonEventTypes) {
-          eventTypesBuilder.add(convertEventType(jsonEventType));
-        }
-        this.eventTypes = eventTypesBuilder.build();
-      }
-      Double samplingRate = JsonUtil.getNumberAsDouble(config, "global_trace_sampling_rate");
-      if (samplingRate == null) {
-        this.sampler = Samplers.probabilitySampler(0.0);
-      } else {
-        checkArgument(
-            samplingRate >= 0.0 && samplingRate <= 1.0,
-            "'global_trace_sampling_rate' needs to be between [0.0, 1.0]");
-        // Using alwaysSample() instead of probabilitySampler() because according to
-        // {@link io.opencensus.trace.samplers.ProbabilitySampler#shouldSample}
-        // there is a (very) small chance of *not* sampling if probability = 1.00.
-        if (1 - samplingRate < EPSILON) {
-          this.sampler = Samplers.alwaysSample();
-        } else {
-          this.sampler = Samplers.probabilitySampler(samplingRate);
-        }
-      }
-      Map<String, ?> rawCustomTags = JsonUtil.getObject(config, "custom_tags");
-      if (rawCustomTags != null) {
-        ImmutableMap.Builder<String, String> builder = new ImmutableMap.Builder<>();
-        for (Map.Entry<String, ?> entry: rawCustomTags.entrySet()) {
-          checkArgument(
-                  entry.getValue() instanceof String,
-                  "'custom_tags' needs to be a map of <string, string>");
-          builder.put(entry.getKey(), (String) entry.getValue());
-        }
-        customTags = builder.build();
-      }
+    checkArgument(config != null, "Invalid configuration");
+    if (config.isEmpty()) {
+      clientLogFilters = Collections.emptyList();
+      serverLogFilters = Collections.emptyList();
+      customTags = Collections.emptyMap();
+      return;
+    }
+    projectId = fetchProjectId(JsonUtil.getString(config, "project_id"));
+
+    Map<String, ?> rawCloudLoggingObject = JsonUtil.getObject(config, "cloud_logging");
+    if (rawCloudLoggingObject != null) {
+      enableCloudLogging = true;
+      ImmutableList.Builder<LogFilter> clientFiltersBuilder = new ImmutableList.Builder<>();
+      ImmutableList.Builder<LogFilter> serverFiltersBuilder = new ImmutableList.Builder<>();
+      parseLoggingObject(rawCloudLoggingObject, clientFiltersBuilder, serverFiltersBuilder);
+      clientLogFilters = clientFiltersBuilder.build();
+      serverLogFilters = serverFiltersBuilder.build();
+    }
+
+    Map<String, ?> rawCloudMonitoringObject = JsonUtil.getObject(config, "cloud_monitoring");
+    if (rawCloudMonitoringObject != null) {
+      enableCloudMonitoring = true;
+    }
+
+    Map<String, ?> rawCloudTracingObject = JsonUtil.getObject(config, "cloud_tracing");
```

Comment:
Looks like in the spec, we are expecting this to be named `cloud_trace` instead?

Reply:
Yes, @stanley-cheung is right. This will need to be changed - provided the Go implementation uses `"cloud-trace"`

Reply:
Thanks for catching it @stanley-cheung. Updated to `cloud_trace`.